pkg/apis/v1/nodepool_conversion.go

```diff
@@ -46,6 +46,7 @@ func (in *NodePool) ConvertTo(_ context.Context, to apis.Convertible) error {
 	// Remove the annotations from the v1beta1 NodeClaim on the convert back
 	delete(v1beta1NP.Annotations, KubeletCompatibilityAnnotationKey)
 	delete(v1beta1NP.Annotations, NodeClassReferenceAnnotationKey)
+	delete(v1beta1NP.Annotations, StoredVersionMigratedKey)
```

Comment:
Can you add a comment on why we do this just so we don't forget in the future if we need